luaui/Widgets/gfx_DrawUnitShape_GL4.lua

```diff
@@ -245,44 +265,49 @@ local function DrawUnitGL4(unitID, unitDefID, px, py, pz, rotationY, alpha, team
-			px, py, pz, rotationY,
-			alpha, 0, teamcoloroverride,highlight ,
-			teamID, 0,
-			0,0,0,0
-		},
-		uniqueID,
+	
+	instanceCache[1], instanceCache[2], instanceCache[3], instanceCache[4] = px, py, pz, rotationY
+	instanceCache[5], instanceCache[6], instanceCache[7], instanceCache[8] = alpha, 0, teamcoloroverride, highlight
+	instanceCache[9] = teamID
+	
+	local elementID = pushElementInstance(
+		DrawUnitVBOTable,
+		instanceCache,
+		updateID,
 		true,
 		nil,
 		unitID,
 		"unitID")
-	return uniqueID
+	return updateID
 end
 
-
-local function DrawUnitShapeGL4(unitDefID, px, py, pz, rotationY, alpha, teamID, teamcoloroverride, highlight)
-	-- Documentation for DrawUnitShapeGL4:
-	--	unitDefID: which unitDef do you want to draw
-	-- px, py, py: where in the world to do you want to draw it
-	-- rotationY: Angle in radians on how much to rotate the unit around Y,
-		-- 0 means it faces south, (+Z),
-		-- pi/2 points west (-X)
-		-- -pi/2 points east
-	-- alpha: the transparency level of the unit
-	-- teamID: which teams teamcolor should this unit get, leave nil if you want to keep the original teamID
-	-- teamcoloroverride: much we should mix the teamcolor into the model color [0-1]
-	-- highlight: how much we should add a highlighting animation to the unit (blends white with [0-1])
-	-- returns: a unique handler ID number that you should store and call StopDrawUnitGL4(uniqueID) with to stop drawing it
-	-- note that widgets are responsible for stopping the drawing of every unit that they submit!
-	uniqueID = uniqueID + 1
-
+---DrawUnitShapeGL4(unitDefID, px, py, pz, rotationY, alpha, teamID, teamcoloroverride, highlight, updateID, ownerID)
+---note that widgets are responsible for stopping the drawing of every unit that they submit! They may use RemoveMyDraws(ownerID) 
+---@param unitDefID number which unitDef do you want to draw
+---@param px number where in the world to do you want to draw it
+---@param py number where in the world to do you want to draw it
+---@param pz number where in the world to do you want to draw it
+---@param rotationY number Angle in radians on how much to rotate the unit around Y, 0 means it faces south, (+Z), pi/2 points west (-X) -pi/2 points east
+---@param alpha number optional the transparency level of the unit
+---@param teamID number optional which teams teamcolor should this unit get, leave nil if you want to keep the original teamID
+---@param teamcoloroverride optional number much we should mix the teamcolor into the model color [0-1]
+---@param highlight number optional how much we should add a highlighting animation to the unit (blends white with [0-1])
+---@param updateID number optional specify the previous uniqueID if you want to update it
+---@param ownerID any optional unique identifier so that widgets can batch remove all of their own stuff
+---@return uniqueID number a unique handler ID number that you should store and call StopDrawUnitGL4(uniqueID) with to stop drawing it
```

Comment:
Might just be a quirk of my IDE, but some of this luadocs stuff is not formatted correctly, like putting `optional` as a type or providing a name to the return value 
![image](https://github.com/beyond-all-reason/Beyond-All-Reason/assets/3493638/9d9a6c3c-39ab-4931-b9e8-0078a009e897)